test/metabase/api/native_query_snippet_test.clj

```diff
@@ -0,0 +1,147 @@
+(ns metabase.api.native-query-snippet-test
+  "Tests for /api/native-query-snippet endpoints."
+  (:require [clojure
+             [string :as str]
+             [test :refer :all]]
+            [metabase.models
+             [database :refer [Database]]
+             [native-query-snippet :refer [NativeQuerySnippet]]
+             [permissions :as perms]
+             [permissions-group :as group]]
+            [metabase.test.data.users :refer [fetch-user user->client]]
+            [toucan.util.test :as tt])
+  (:import java.time.LocalDateTime))
+
+(def ^:private test-snippet-fields [:content :creator_id :database_id :description :name])
+(def ^:private rasta (fetch-user :rasta))
```

Comment:
you definitely shouldn't do this. `fetch-user` fetches the user from the application DB. It does initialize the DB if needed as a side-effect of calling it but we shouldn't be initializing the app DB as a side effect of loading a namespace. 

That shouldn't happen until tests start running.

Reply:
I used to have the test runner throw an Exception if you were doing stuff with the application DB or what-not as a side-effect of loading a namespace but I'm not 100% sure how to do it with `clojure.test` (another limitation of the framework). I'll have to see if I can do something like swap out `run-all-tests` to get this check in place

Reply:
Agreed and fixed.

> I used to have the test runner throw an Exception if you were doing stuff with the application DB or what-not as a side-effect of loading a namespace but I'm not 100% sure how to do it with `clojure.test`

Would it perhaps be possible to overload `clojure.test`'s runner entry point (kinda like how you're currently "hijacking" `expect`) and replace it with one that errors when it finds an initialized DB?

Reply:
I wouldn't describe what I'm doing with `expect` as hijacking so much as it is providing an alternative implementation -- we don't have a dependency on the `expectations` library anymore, just a namespace with an `expect` macro that expands to a `deftest` form.

It is totally possible to swap out the test runner functions, which would be hijacking. The test runner itself loads the namespaces so it's a little tricky to tell when we're actually starting the test run vs just loading the namespace. I played around with it for a half hour or so and didn't figure it out yet but with some more time should be able to figure it out. 

It might be easier just to write a lein command that loads all the test namespaces and checks whether anything is initialized and then run that as a separate linter